src/BasePlatform.ts

```diff
@@ -508,4 +508,24 @@ export default abstract class BasePlatform {
      * Begin update polling, if applicable
      */
     public startUpdater(): void {}
+
+    /**
+     * Checks if the current session is lock-free, i.e., no other instance is holding the session lock.
+     * Platforms that support session locking should override this method.
+     * @returns {Promise<boolean>} True if the session is lock-free, false otherwise.
+     */
+    public async checkSessionLockFree(): Promise<boolean> {
+        return false;
+    }
+
+    /**
+     * Attempts to acquire a session lock for this instance.
+     * If another instance is detected, calls the provided callback.
+     * Platforms that support session locking should override this method.
+     * @param _onNewInstance Callback to invoke if a new instance is detected.
+     * @returns {Promise<boolean>} True if the lock was acquired, false otherwise.
+     */
+    public async getSessionLock(_onNewInstance: () => Promise<void>): Promise<boolean> {
+        return false;
+    }
```

Comment:
Should these maybe be abstract given none of the subclasses inherit their implementations?